### PR TITLE
Feature/jfshen

### PR DESCRIPTION
**Pull request by jfshenuw**

### CODE REVIEW

File: acs.py [hunk 1]
```diff
@@ -220,6 +220,28 @@ def createOrUpdateFilePlan(acs, filePlan):
             if 'children' in rc:
                 createOrUpdateChildCategories(acs, category['id'], rc['children']);
 
+#############################################
+# create app user
+def createAppUser(acs, user):
```
Comment (agagne) [inline acs.py]: can we rename this 'createAdminAppUser'

File: acs.py [hunk 2]
```diff
@@ -248,6 +270,11 @@ def main():
             acs.createGroup(adminGroup, adminGroup)
         acs.addGroupMember('ALFRESCO_ADMINISTRATORS', adminGroup)
 
+    # create app users
+    if conf and conf['appUsers']:
```
Comment (agagne) [inline acs.py]: can we name this adminAppUsers

File: acs.yml.example [hunk 3]
```diff
@@ -4,6 +4,10 @@ default:
   url: <Alfresco Repository URL, e.g. 'https://localhost:8080'>
   rootGroup: my_root_group
   adminGroup: my_admin_group
+  appUsers:
```
Comment (agagne) [inline acs.yml.example]: can we name this 'adminAppUsers'

File: migrate.py [hunk 4]
```diff
@@ -313,40 +345,91 @@ def __prettify_xml(self, elem):
 
 class HdaTranslator:
     def __init__(self, content_model_definition_file, content_model_profile, csv_file,
-                 hda_input_file, number_of_docs_to_process, print_to_screen, xml_output_directory,
-                 should_validate_field_value, sample_files=None):
+                 wcc_archives_input_dir, number_of_docs_to_process, print_to_screen, output_directory,
+                 should_validate_field_value, seq1, seq2, countFile, sample_files_dir=None):
```
Comment (agagne) [inline migrate.py]: can seq1 & seq2 be given more descriptive names?

File: acs.yml.example (hunk 3 above)
Comment (jfshenuw) [reply acs.yml.example]: renamed.

File: acs.py (hunk 2 above)
Comment (jfshenuw) [reply acs.py]: renamed

File: acs.py (hunk 1 above)
Comment (jfshenuw) [reply acs.py]: renamed

File: migrate.py [hunk 5]
```diff
@@ -313,40 +345,91 @@ def __prettify_xml(self, elem):
...
+
+        count_file_dir = self.output_directory + '/count_files'
+        if not os.path.exists(count_file_dir):
+            os.makedirs(count_file_dir)
+
+        prev_count_file = self.countFile
+        global name_field_value_count
+        for f in sorted(os.listdir(self.wcc_archives_input_dir), key=take_seq):
+            if f.endswith('.hda') and f != 'docmetadefinition.hda':
+                seqno = f.split('~')[1].split('.')[0]
```
Comment (agagne) [inline migrate.py]:
```suggestion
                seqno = take_seq(f)
```

File: migrate.py [hunk 6]
```diff
@@ -313,40 +345,91 @@ def __prettify_xml(self, elem):
...
+                iSeqno = int(seqno)   # seqno is always an integer
```
Comment (agagne) [inline migrate.py]: unnecessary variable if you use take_seq function to aquire the seqno as an int

File: migrate.py (hunk 4 above)
Comment (jfshenuw) [reply migrate.py]: We could use start_sequence_number and end_sequence_number, which are fairly long names. This script has very limited lifespan, and we probably want to spend minimal time on it, provided it produces the correct results.

Comment (agagne) [reply migrate.py]: start_sequence_number and end_sequence_number seem fine to me, it'll make the code easier to read than seq1 seq2 ..

**Pull request merged**